lighthouse-core/lib/file-namer.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/**
+ * Generate a filenamePrefix of hostname_YYYY-MM-DD_HH-MM-SS
+ * Date/time uses the local timezone, however Node has unreliable ICU
+ * support, so we must construct a YYYY-MM-DD date format manually. :/
+ * @param {!Object} results
+ * @returns string
+ */
+function getFilenamePrefix(results) {
+  // eslint-disable-next-line no-undef
+  const hostname = new (URLConstructor || URL)(results.url).hostname;
```

Comment:
`url-shim` already guards against this. Any way we could rely on that and avoid this (and the `require()` below)? Since `file-namer` isn't included via script tag it seems like it's doable.